parsers/CZ.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from typing import List, Optional, Union
+
+from bs4 import BeautifulSoup
+
+# The request library is used to fetch content through HTTP
+from requests import Session
+
+from parsers.lib.exceptions import ParserException
+
+# please try to write PEP8 compliant code (use a linter). One of PEP8's
+# requirement is to limit your line length to 79 characters.
+
+
+translate_table_gen = {
+    "TPP": "coal",  # coal
+    "CCGT": "gas",  # gas and steem gas
+    "NPP": "nuclear",  # Nuclear
+    "HPP": "hydro",  # Water
+    "PsPP": "hydro",  # Pump Water storage
+    "AltPP": "biomass",  # Alternative
+    "ApPP": "unknown",  # factory
+    "PVPP": "solar",  # photovoltaic
+    "WPP": "wind",  # wind
+    "unknown": "unknown",
+}
+translate_table_dist = {
+    "SEPS": "SVK",
+    "APG": "AT",
+    "PSE": "PL",
+    "TenneT": "DE",
+    "50HzT": "DE",
+}
+
+url = "https://www.ceps.cz/_layouts/CepsData.asmx"
+
+
+def get_mapper(xmlload):
+    series = xmlload.find("series")
+    mapping = {}
+    for tag in series:
+        generator = tag["name"].replace(" [MW]", "")
+        mapping[generator] = tag["id"]
+
+    return mapping
+
+
+def make_request(session, payload, zone_key):
+    headers = {
+        "Content-Type": "application/soap+xml; charset=utf-8",
+        "Content-Length": "1",
+    }
+
+    r = session or Session()
+    res = r.post(url, headers=headers, data=payload)
+    assert res.status_code == 200, (
+        "Exception when fetching production for "
+        "{}: error when calling url={}".format(zone_key, url)
+    )
```

Comment:
```suggestion
def make_request(session: Session, payload: str, zone_key: str):
    headers = {
        "Content-Type": "application/soap+xml; charset=utf-8",
        "Content-Length": "1",
    }

    res: Response = session.post(url, headers=headers, data=payload)
    assert res.status_code == 200, (
        f"Exception when fetching production for {zone_key}: error when calling {url}"
    )
```
Let's clean this up a bit and use the newest syntax.